File: service_booking_management/service_booking_management/report/wellness_services_summary/wellness_services_summary.py
import frappe
from frappe.utils import format_datetime


def execute(filters=None):
    columns = get_columns(filters)
    data = get_data(filters)
    return columns, data


def get_columns(filters):
    columns = [
        {
            "label": "Service Booking ID",
            "fieldname": "name",
            "fieldtype": "Link",
            "options": "Service Booking",
            "width": 200
        },
        {
            "label": "Customer",
            "fieldname": "customer_name",
            "fieldtype": "Data",
            "width": 200
        },
        {
            "label": "Service",
            "fieldname": "service_type",
            "fieldtype": "Data",
            "width": 150
        }
    ]

    # Only include 'Other Services' column if filter is not applied or is set to "Others"
    if not filters or filters.get("service_type") == "Others":
        columns.append({
            "label": "Other Services",
            "fieldname": "other_services",
            "fieldtype": "Small Text",
            "width": 350
        })

    columns.extend([
        {
            "label": "Status",
            "fieldname": "workflow_state",
            "fieldtype": "Data",
            "width": 120
        },
        {
            "label": "Date/Time",
            "fieldname": "preferred_datetime",
            "fieldtype": "Data",
            "width": 180
        }
    ])

    return columns


def get_data(filters):
    fields = ["name", "customer_name", "service_type",
              "workflow_state", "preferred_datetime"]

    # Only fetch 'other_services' if needed
    if not filters or filters.get("service_type") == "Others":
        fields.append("other_services")

    service_booked_records = frappe.get_all(
        "Service Booking",
        fields=fields,
        filters=filters
    )

    for row in service_booked_records:
        if row.get("preferred_datetime"):
            row["preferred_datetime"] = format_datetime(
                # e.g. 3, Jul, 2025 | 12:30
                row["preferred_datetime"], "d, MMM, yyyy | hh:mm"
            )

    return service_booked_records

File: service_booking_management/service_booking_management/report/wellness_services_summary/wellness_services_summary.js
// Copyright (c) 2025, sessions.Abhimanyu and contributors
// For license information, please see license.txt

frappe.query_reports["Wellness Services Summary"] = {
	filters: [
		{
			fieldname: "service_type",
			label: "Service Type",
			fieldtype: "Select",
			options: ["", "Therapy", "Spa", "Others"],
		},
		{
			fieldname: "workflow_state",
			label: "Status",
			fieldtype: "Select",
			options: ["", "Requested", "Approved", "Completed"],
		},
	],
	onload: function (report) {
		report.page.add_inner_button("New Service Booking", function () {
			frappe.set_route("form", "Service Booking", "new");
		});
	},
	formatter: function (value, row, column, data, default_formatter) {
		value = default_formatter(value, row, column, data);

		if (column.fieldname === "workflow_state") {
			const colorMap = {
				Requested: "orange",
				Approved: "blue",
				Completed: "green",
			};

			const color = colorMap[data.workflow_state] || "gray";

			return `<span class="indicator-pill ${color} no-indicator-dot ellipsis">${data.workflow_state}</span>`;
		}

		return value;
	},
};
